provider/pkg/provider/deployment_settings.go

```diff
@@ -349,11 +351,8 @@ func toOperationContext(inputMap resource.PropertyMap) *pulumiapi.OperationConte
 		evInput := ocInput["environmentVariables"].ObjectValue()
 
 		for k, v := range evInput {
-			if v.IsSecret() {
-				ev[string(k)] = apitype.SecretValue{Secret: true, Value: v.SecretValue().Element.StringValue()}
-			} else {
-				ev[string(k)] = apitype.SecretValue{Secret: false, Value: v.StringValue()}
-			}
+			// This hardcodes all env vars created via provider as secret. Schema change would be needed for optional secrecy
+			ev[string(k)] = apitype.SecretValue{Secret: true, Value: v.StringValue()}
```

Comment:
Why is this change necessary? Previously if the value passed in was a pulumi secret we would set it as a secret. Otherwise it would not be marked secret. Looks like we changed this to mark everything as secret, but it's not clear why.

Reply:
This change was not strictly necessary, but the way things were there was no way to make a secret env var. (Correct me if I'm wrong, env var map was just a map of string to string). 

So I opted for secret always vs non-secret always. As I mentioned in description, it would be great to come back and change the schema and actually allow both secret and non-secret env vars.

Reply:
Yeah I think you're mistaken. With the current implementation, if you mark something as secret the pulumi way (pulumi.Secret or config.requireSecret) it will set the the env var as secret. It's not in the schema but we do already respect secretness. Recommend reading this doc and trying it out with this change reverted. https://www.pulumi.com/docs/concepts/secrets/

Reply:
This test has some examples: https://github.com/pulumi/pulumi-pulumiservice/blob/main/examples/ts-deployment-settings/index.ts

Reply:
Just tried it out, `Output<String>` is black magic 🤯 

I will need to adjust parsing in `toOperationContext`, because adding secrets turns it into below struct, but other than that it should work I think. 
```
{
	struct_value: {
		fields:{
			key:"environmentVariables"
			value: {
				struct_value:{
					fields:{
						key:"4dabf18193072939515e22adb298388d"
						value:{
							string_value:"1b47061264138c4ac30d75fd1eb44270"
						}
					}
					fields:{
						key:"value"
						value:{
							struct_value:{
								fields:{
									key:"IaroEnvVarKey"
									value:{
										string_value:"[secret]"
									}
								}
							}
						}
					}
				}
			}
		}
	}
} 
```

I will figure out the parsing and update PR, likely next week.